api/medical/views.py

```diff
@@ -170,3 +172,78 @@ def get(self, request, *args, **kwargs):
         data = Staff.objects.filter(id=pk).first()
         serialized_json = StaffSerializer(data)
         return Response(data=serialized_json.data)
+
+
+class AppointmentView(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request):
+        doctor = request.POST.get("doctor", None)
+        doctorinstance = Doctor.objects.filter(id=doctor).first()
+        user = request.user
+        instance = Appointment.objects.create(
+            doctor=doctorinstance,
+            patient=user
+        )
+        instance.save()
+        serinstance = AppointmentSerializer(instance)
+        return Response(serinstance.data)
+
+    def get(self, request):
+        user = request.user
+        appointments = Appointment.objects.filter(patient=user)
+        serializer = AppointmentSerializer(appointments, many=True)
+        return Response(serializer.data)
+
+
+class DoctorAppointmentView(APIView):
+    permission_classes = (IsMedicalManager,)
+
+    def get(self, request):
+        doctor = request.user
+        appointments = Appointment.objects.filter(doctor=doctor)
+        serializer = AppointmentSerializer(appointments, many=True)
+        return Response(serializer.data)
+
+    def post(self, request):
+        id = request.POST.get("id", None)
+        accepted = request.POST.get("accepted", None)
+        appointment = Appointment.objects.filter(id=id).first()
+        if accepted == "true":
+            date = request.POST.get("date", None)
+            time = request.POST.get("time", None)
+            appointment.date = date
+            appointment.time = time
+            appointment.accepted = True
+        else:
+            reason = request.POST.get("reason", None)
+            appointment.reason = reason
+            appointment.accepted = False
+        appointment.save()
+        serinstance = AppointmentSerializer(appointment)
```

Comment:
ditto everywhere

Reply:
unable to understand the issue can you please breif